components/Common/OramaSearchBar/index.tsx

```diff
@@ -0,0 +1,64 @@
+import { create, load } from '@orama/orama';
+import { useTheme } from 'next-themes';
+import { useCallback, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { OramaLogoDark, OramaLogoLight } from './OramaLogo';
+import styles from './index.module.scss';
+import { identityDocumentMapper, performSearch } from './search';
+import SearchBar from '../SearchBar';
+
+import type { FC } from 'react';
+import type { OramaSearchBarProps } from './search';
+import type { SearchFunction } from '../../../types';
+
+const Footer: FC = () => {
+  const { resolvedTheme: theme } = useTheme();
+
+  const isDark = theme === 'dark';
+
+  return (
+    <div className={styles.oramaSearchBarFooter}>
+      <span className={styles.oramaSearchBarFooterLabel}>
+        <FormattedMessage id="components.oramaSearchBar.search.poweredBy" />
+      </span>
+      <a
+        href="http://oramajs.io"
+        target="_blank"
+        rel="noopener noreferrer"
+        className={styles.oramaSearchBarFooterLink}
+      >
+        {isDark ? (
+          <OramaLogoDark className={styles.oramaSearchBarFooterLogo} />
+        ) : (
+          <OramaLogoLight className={styles.oramaSearchBarFooterLogo} />
+        )}
+      </a>
+    </div>
+  );
+};
+
+const OramaSearchBar: FC<OramaSearchBarProps> = ({
+  schema,
+  index,
+  documentMapper,
+}: OramaSearchBarProps) => {
+  const [searchFunction, setSearchFunction] = useState<SearchFunction>();
+
+  // Create a setup function to deserialize the database
+  const setup = useCallback(async () => {
```

Comment:
Is there any reason why we're passing the setup function to the child instead of just  doing the setup here. It feels more complicated.

Reply:
I designed `SearchBar` to be backend agnostic, therefore it accepts a `setup` function to prepare (and thus not allow searching before setup has completed.)
If we do as you suggested, then `useCallback` will become a `useEffect`, and then I'll have to change SearchBar to accept a boolean for `isLoading`. I see little benefit here as the setup part is separated from the component anyway.